src/main/java/org/prebid/server/bidder/adf/AdfBidder.java

```diff
@@ -1,56 +1,162 @@
+            onError.accept(BidderError.badInput(bidderErrorMessage));
+            return null;
+        }
+
+        return imp.withTagid(optionalAdfImp.orElseThrow().getMid());
+    }
+
+    private HttpRequest<BidRequest> makeRequest(BidRequest bidRequest, List<Imp> imps) {
+        final BidRequest outgoingRequest = bidRequest.withImp(imps);
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
                 .build();
     }
 
     @Override
-    protected BidType getBidType(Bid bid, List<Imp> imps) {
-        final String impId = bid.getImpid();
-        final BidType bidType = getExtBidPrebidType(bid.getExt());
-        if (bidType != null) {
-            return bidType;
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = parseBidResponse(httpCall);
+            final List<BidderError> errors = new ArrayList<>();
+            final List<BidderBid> bidderBids = extractBids(bidResponse, errors::add);
+            return Result.of(bidderBids, errors);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
-
-        throw new PreBidException(String.format("Failed to parse bid %s mediatype", impId));
     }
 
-    private BidType getExtBidPrebidType(ObjectNode bidExt) {
-        final ExtBidPrebid extBidPrebid = getExtBidPrebid(bidExt);
-        return extBidPrebid != null ? extBidPrebid.getType() : null;
+    private BidResponse parseBidResponse(HttpCall<BidRequest> httpCall) throws DecodeException {
+        return mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
     }
 
-    private ExtBidPrebid getExtBidPrebid(ObjectNode bidExt) {
-        if (bidExt == null || !bidExt.hasNonNull(PREBID_EXT)) {
-            return null;
+    private List<BidderBid> extractBids(BidResponse bidResponse, Consumer<BidderError> onError) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+
+        if (bidResponse.getCur() == null) {
+            onError.accept(BidderError.badServerResponse("Currency is null."));
+            return Collections.emptyList();
         }
+
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .filter(Objects::nonNull)
+                .map(bid -> makeBidderBid(bid, bidResponse.getCur(), onError))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
```

Comment:
Pls check other Bidders code style, like between bidder, and update your code style to be as much similar as possible